indra/newview/llavatarpropertiesprocessor.cpp

```diff
@@ -167,19 +159,16 @@ void LLAvatarPropertiesProcessor::sendAvatarPropertiesRequestMessage(const LLUUI
     gAgent.sendReliableMessage();
 }
 
-void LLAvatarPropertiesProcessor::initAgentProfileCapRequest(const LLUUID& avatar_id, const std::string& cap_url)
+void LLAvatarPropertiesProcessor::initAgentProfileCapRequest(const LLUUID& avatar_id, const std::string& cap_url, EAvatarProcessorType type)
 {
-    addPendingRequest(avatar_id, APT_PROPERTIES);
-    addPendingRequest(avatar_id, APT_PICKS);
-    addPendingRequest(avatar_id, APT_GROUPS);
-    addPendingRequest(avatar_id, APT_NOTES);
+    addPendingRequest(avatar_id, type);
```

Comment:
This is somewhat ineffective, if something sends an apt_properties request then user opens picks for example, it will trigger initAgentProfileCapRequest again (unlikely, but possible), viewer will be requesting and getting same data twice. Instead of should mark all the data as pending so that if user opens a mini profile then picks, both will get updated via same message instead of requiring two.

Reply:
By this point most APT_* need to be merged into a single one with old system disabled where possible.